src/lib.rs

```diff
@@ -400,6 +400,307 @@ pub extern "C" fn DATE_AND_TIME_TO_TIME_OF_DAY(input: i64) -> i64 {
+/// This operator returns the value of adding up TOD and TIME.
+/// Panic on onverflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn ADD_TOD_TIME(in1: i64, in2: i64) -> i64 {
+    add_datetime_time(in1, in2)
+}
+
+/// .
+/// This operator returns the value of adding up DT and TIME.
+/// Panic on overflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn ADD_DT_TIME(in1: i64, in2: i64) -> i64 {
+    add_datetime_time(in1, in2)
+}
+
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn add_datetime_time(in1: i64, in2: i64) -> i64 {
+    chrono::Utc
+        .timestamp_millis(in1)
+        .checked_add_signed(chrono::Duration::nanoseconds(in2))
+        .unwrap()
+        .timestamp_millis()
+}
+
+/// .
+/// This operator produces the subtraction of two TIME operands
+/// Panic on overflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn SUB_TIME(in1: i64, in2: i64) -> i64 {
+    chrono::Duration::milliseconds(in1)
+        .checked_sub(&chrono::Duration::milliseconds(in2))
+        .unwrap()
+        .num_milliseconds()
+}
+
+/// .
+/// This operator produces the subtraction of two DATE operands
+/// Panic on overflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn SUB_DATE_DATE(in1: i64, in2: i64) -> i64 {
+    sub_datetimes(in1, in2)
+}
+
+/// .
+/// This operator produces the subtraction of TOD and TIME
+/// Panic on overflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn SUB_TOD_TIME(in1: i64, in2: i64) -> i64 {
+    sub_datetime_duration(in1, in2)
+}
+
+/// .
+/// This operator produces the subtraction of two TOD operands
+/// Panic on overflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn SUB_TOD_TOD(in1: i64, in2: i64) -> i64 {
+    sub_datetimes(in1, in2)
+}
+
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn sub_datetimes(in1: i64, in2: i64) -> i64 {
+    chrono::Utc
+        .timestamp_millis(in1)
+        .signed_duration_since(chrono::Utc.timestamp_millis(in2))
+        .num_nanoseconds()
+        .unwrap()
+}
+
+/// .
+/// This operator produces the subtraction of DT and TIME
+/// Panic on overflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn SUB_DT_TIME(in1: i64, in2: i64) -> i64 {
+    sub_datetime_duration(in1, in2)
+}
+
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn sub_datetime_duration(in1: i64, in2: i64) -> i64 {
+    chrono::Utc
+        .timestamp_millis(in1)
+        .checked_sub_signed(chrono::Duration::nanoseconds(in2))
+        .unwrap()
+        .timestamp_millis()
+}
+
+/// .
+/// This operator produces the subtraction of two DT operands
+/// Panic on overflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn SUB_DT_DT(in1: i64, in2: i64) -> i64 {
+    sub_datetimes(in1, in2)
+}
+
+/// .
+/// Multiply TIME/LTIME with ANY_SIGNED_INT
+/// Panic on overflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn CHECKED_MUL_SIGNED(in1: i64, in2: i64) -> i64 {
```

Comment:
are the CHECKED_... methods being called in ST? are they in the norm, if not, just keep them to rust for now

Reply:
yes they are called in the MUL_TIME/MUL_LTIME functions in "date_time_numeric_functions.st"
no they are not in the norm, i made them as helper functions to reduce code duplication
should we keep them like this because they are called in ST ?

Reply:
Yes if we used them in ST we can keep them